code/game/machinery/genpop.dm

```diff
@@ -0,0 +1,372 @@
+			return
+		if(!I || !I.registered_name)
+			return
+		//Sec officers can always open the lockers. Bypass the ID setting behaviour.
+		//Buuut, if they're holding a prisoner ID, that takes priority.
+		if(allowed(user) && !istype(I, /obj/item/card/id/prisoner))
+			locked = !locked
+			update_icon()
+			return TRUE
+		//Handle setting a new ID.
+		if(!registered_name)
+			if(istype(I, /obj/item/card/id/prisoner)) //Don't claim the locker for a sec officer mind you...
+				var/obj/item/card/id/prisoner/P = I
+				if(P.assigned_locker)
+					to_chat(user, "<span class='notice'>This ID card is already registered to a locker.</span>")
+					return FALSE
+				P.assigned_locker = src
+				registered_name = I.registered_name
+				desc = "Assigned to: [I.registered_name]."
+				say("Locker sealed. Assignee: [I.registered_name]")
+
+				playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, 0)
+				set_light(1,1,COLOR_RED)
+				locked = TRUE
+				update_icon()
+			else
+				to_chat(user, "<span class='danger'>Invalid ID. Only prisoners / officers may use these lockers.</span>")
+			return FALSE
+		//It's an ID, and is the correct registered name.
+		if(istype(I) && (registered_name == I.registered_name))
+			var/obj/item/card/id/prisoner/ID = I
+			//Not a prisoner ID.
+			if(!istype(ID))
+				return FALSE
+			if(ID.served_time < ID.sentence)
+				playsound(loc, 'nsv13/sound/effects/computer/alarm_3.ogg', 80)
+				say("DANGER: PRISONER HAS NOT COMPLETED SENTENCE. AWAIT SENTENCE COMPLETION. COMPLIANCE IS IN YOUR BEST INTEREST.")
+				return FALSE
+			visible_message("<span class='warning'>[user] slots [I] into [src]'s ID slot, freeing its contents!</span>")
+			registered_name = null
+			desc = initial(desc)
+			locked = FALSE
+			update_icon()
+			set_light(1,1,COLOR_GREEN)
+			qdel(I)
+			playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, 0)
+			return
+		else
+			to_chat(user, "<span class='danger'>Access Denied.</span>")
+	else
+		return ..()
+
+/obj/machinery/turnstile/Initialize()
+	. = ..()
+	icon_state = "turnstile"
+
+/obj/machinery/turnstile/CanAtmosPass(turf/T)
+	return TRUE
+
+/obj/machinery/turnstile/CanPass(atom/movable/mover, turf/target)
+	if(istype(mover) && (mover.pass_flags & PASSGLASS))
+		return TRUE
+	if(!isliving(mover))
+		return TRUE
+	var/allowed = allowed(mover)
+	//Sec can drag you out unceremoniously.
+	if(!allowed && mover.pulledby)
+		allowed = allowed(mover.pulledby)
+	if(get_dir(loc, target) == dir || allowed) //Make sure looking at appropriate border
+		flick("operate", src)
+		playsound(src,'sound/items/ratchet.ogg',50,0,3)
+		return TRUE
+	else
+		flick("deny", src)
+		playsound(src,'sound/machines/deniedbeep.ogg',50,0,3)
+		return FALSE
+
+//Officer interface.
+/obj/machinery/genpop_interface
+	name = "Prisoner Management Interface"
+	icon = 'icons/obj/status_display.dmi'
+	icon_state = "frame"
+	desc = "An all-in-one interface for officers to manage prisoners!"
+	req_access = list(ACCESS_SECURITY)
```

Comment:
I think we have ACCESS_HOLDINGCELLS or something instead

Reply:
If that exists it is extremely snowflakey

Reply:
It's `ACCESS_BRIG` actually

Reply:
<!--StartFragment-->
2 | Brig | Brig cells+timers, permabrig, gulag+gulag shuttle, prisoner management console
-- | -- | --


<!--EndFragment-->

https://tgstation13.org/wiki/Guide_to_door_access

Reply:
> 2	Brig	Brig cells+timers, permabrig, gulag+gulag shuttle, prisoner management console
> https://tgstation13.org/wiki/Guide_to_door_access

Linking to TG's wiki instead of our own 😩